src/blob-store/index.js

```diff
@@ -274,6 +277,55 @@ class PretendCorestore {
     }
   }
 
+  /**
+   * Write blobs for provided variants of a file
+   * @param {{ original: string, preview?: string, thumbnail?: string }} filepaths 
+   * @param {{ mimeType?: string }} metadata
+   * @returns {Promise<{ original: Omit<BlobId, 'driveId'>, preview?: Omit<BlobId, 'driveId'>, thumbnail?: Omit<BlobId, 'driveId'> }>}
+   */
+  async create(filepaths, metadata) {
+    const { original, preview, thumbnail } = filepaths
+    const { mimeType } = metadata
+
+    const originalBlobId = await writeFile(original)
+    const previewBlobId = preview ? await writeFile(preview) : null
+    const thumbnailBlobId = thumbnail ? await writeFile(thumbnail) : null
+
+    const blobIds = /** @type {{ original: Omit<BlobId, 'driveId'>, preview?: Omit<BlobId, 'driveId'>, thumbnail?: Omit<BlobId, 'driveId'> }} */({
+      original: originalBlobId,
+    })
+
+    if (previewBlobId) blobIds.preview = previewBlobId
+    if (thumbnailBlobId) blobIds.thumbnail = thumbnailBlobId
+
+    return blobIds
+
+    /**
+     * @param {string} filename 
+     * @returns {Promise<Omit<BlobId, 'driveId'>>}
+     */
+    function writeFile (filename) {
+      const [type, variant, name] = this.splitPath(filename)
+      return new Promise((resolve, reject) => {
+        fs.createReadStream(filename)
+          .pipe(this.createWriteStream({ type, variant, name }, { metadata: { mimeType } }))
+          .on('error', reject)
+          .on('finish', () => {
+            resolve({ type, variant, name })
+          })
+      })
+    }
+  }
+
+  /**
+   * 
+   * @param {string} filepath 
+   * @returns {Omit<BlobId, 'driveId'>}
+   */
+  splitPath (filepath) {
```

Comment:
Is there any need for this to be a public method?

```suggestion
  #splitPath (filepath) {
```